src/Type/WPInterfaceTrait.php

```diff
@@ -133,15 +187,40 @@ protected function get_fields( array $config, TypeRegistry $type_registry ): arr
 		$fields = array_merge( $fields, array_diff_key( $interface_fields, $fields ) );
 
 		foreach ( $fields as $field_name => $field ) {
-			$merged_field_config = $this->inherit_field_config_from_interface( $field_name, $field, $interface_fields );
+			// Get all interfaces that define this field
+			$interfaces_with_field = [];
+			foreach ( $this->getInterfaces() as $interface ) {
+				if ( ! $interface instanceof InterfaceType ) {
+					$interface = $type_registry->get_type( $interface );
+				}
 
-			if ( null === $merged_field_config ) {
-				unset( $fields[ $field_name ] );
-				continue;
+				if ( ! $interface instanceof InterfaceType ) {
+					continue;
+				}
+
+				$interface_fields = $interface->getFields();
+				if ( isset( $interface_fields[ $field_name ] ) ) {
+					$interfaces_with_field[] = $interface;
+				}
 			}
```

Comment:
How is this different from what happens in `::get_fields_from_implemented_interfaces()` ?